lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,309 @@
+
+                 enrollError: function() {
+                     this.$dateDisplayField.find('.fa.fa-spin').removeClass('fa-spin fa-spinner').addClass('fa-close');
+                     this.$dateDisplayField
+                         .append(gettext('There was an error, please reload the page and try again.'));
+                     this.hideVerificationDialog();
+                 },
+
+                 updateEnrollBtn: function() {
+                     /*
+                     This function is invoked on load, on opening the view and on changing the option on the session
+                     selection dropdown. It plays three roles:
+                     1) Enables and disables enroll button
+                     2) Changes text to describe the action taken
+                     3) Formats the confirmation popover to allow for two step authentication
+                     */
+                     var enrollText,
+                         confirmationHTML,
+                         currentSessionId = this.entitlementModel.get('currentSessionId'),
+                         newSessionId = this.$('.session-select').find('option:selected').data('session_id'),
+                         enrollBtnInitial = this.$('.enroll-btn-initial');
+
+                     // Disable the button if the user is already enrolled in that session.
+                     if (currentSessionId === newSessionId) {
+                         enrollBtnInitial.addClass('disabled');
+                         enrollBtnInitial.popover('dispose');
+                         return;
+                     }
+                     enrollBtnInitial.removeClass('disabled');
+
+                     // Update button text specifying if the user is initially enrolling, changing or leaving a session.
+                     if (newSessionId) {
+                         enrollText = currentSessionId ? gettext('Change Session') : gettext('Select Session');
+                     } else {
+                         enrollText = gettext('Leave Current Session');
+                     }
+                     this.$('.enroll-btn-initial').text(enrollText);
+
+                     // Update the button popover text to enable two step authentication.
+                     if (newSessionId) {
+                         confirmationHTML = currentSessionId ?
+                             gettext('Are you sure that you would like to switch session?') + '<br><br>' +
+                             gettext(' Please know that by choosing to switch session you will lose any progress you' +
+                             'have made in this session.') :
+                             gettext('Are you sure that you would like to select this session?');
+                     } else {
+                         confirmationHTML = gettext('Are you sure that you would like to leave this session?') +
+                             '<br><br>' + gettext('Please know that by leaving you will lose any progress you had' +
```

Comment:
It would be cleaner to use CSS instead of ```<br>```

Reply:
Changed